load.c

```diff
@@ -1511,6 +1515,29 @@ rb_f_autoload_p(int argc, VALUE *argv, VALUE obj)
     return rb_mod_autoload_p(argc, argv, klass);
 }
 
+void *
+rb_dln_resolve_symbol(const char* fname, const char* symbol)
+{
+    VALUE handle;
+    VALUE resolved;
+    VALUE path;
+    VALUE fname_str = rb_str_new_cstr(fname);
+
+    resolved = rb_resolve_feature_path((VALUE)NULL, fname_str);
+    if (NIL_P(resolved) || RARRAY_LEN(resolved) != 2) {
+        return NULL;
+    }
+    if (rb_ary_entry(resolved, 0) != ID2SYM(rb_intern("so"))) {
+        return NULL;
+    }
+    path = rb_ary_entry(resolved, 1);
+    handle = rb_hash_aref(ruby_dln_libmap, path);
```

Comment:
`rb_hash_aref` unnecessarily uses the default value.
 
```suggestion
    handle = rb_hash_lookup(ruby_dln_libmap, path);
```